### ISSUE TITLE
Improve focus-trap to optionally aria-disable everything outside the trapped region

### ISSUE THREAD

**Issue by jelbourn** [angular/components#9035]
#9016 adds this behavior to the dialog specifically (so that it is a patch fix), but the underlying focus-trap should be expanded with this feature. This, however, would change how the API for focus-trap is structured and will need to go in a major version.

**Comment by jelbourn** [comment]
I believe this is obsolete with the new focus trap implementation.
